src/app/zap-templates/helper-chip.js

```diff
@@ -65,10 +52,102 @@ function chip_header() {
   */`;
 }
 
+function isString(type) {
+  return type == 'CHAR_STRING' || type == 'OCTET_STRING' || type == 'LONG_CHAR_STRING' || type == 'LONG_OCTET_STRING';
+}
+
+function isNullable(type, isArray) { return isString(type) || isArray; }
+
+function asValueIfNotPresent(type) {
+  var promise = templateUtil.ensureZclPackageId(this)
+                    .then((packageId) => {
+                      switch (type) {
+                      case 'CHAR_STRING':
+                      case 'OCTET_STRING':
+                        return 'NULL'
+                        case 'LONG_CHAR_STRING': case 'LONG_OCTET_STRING': return 'NULL' default: return cHelper.asUnderlyingType
+                            .call(this, type)
+                            .then(zclType => {
+                              switch (zclType) {
+                              case 'uint8_t':
+                                return 'UINT8_MAX'
+                                case 'uint16_t': return 'UINT16_MAX'
+                                case 'uint32_t': return 'UINT32_MAX' default: error =
+                                    'Unhandled underlying type ' + zclType + ' for original type ' + type
+                                throw error
+                              }
+                            })
+                      }
+                    })
+                    .catch((err) => { console.log(err); });
+
+  return templateUtil.templatePromise(this.global, promise)
+}
+function asTypeLength(type) {
+  var promise = templateUtil.ensureZclPackageId(this)
+                    .then((packageId) => {
+                      switch (type) {
+                      case 'CHAR_STRING':
+                      case 'OCTET_STRING':
+                        return '1u'
+                        case 'LONG_CHAR_STRING': case 'LONG_OCTET_STRING': return '2u' default: return cHelper.asUnderlyingType
```

Comment:
Ah. Last time I looked `char_string`,  `octet_string`, `long_char_string`, and `long_octet_string` didn't get an `ATOMIC_SIZE`.  I guess it has been fixed. Will look at it again. Thanks for the heads up.

Reply:
For those type, you're right the database doesn't hold the size. My comment was for the `default` case. Since `asUnderlyingType` provide a level of abstraction we might run into unwanted behaviour

Reply:
Fixed.